cli/tests/pcluster/config/test_config_patch.py

```diff
@@ -845,6 +845,39 @@ def _test_storage(base_conf, target_conf):
     )
 
 
+def _test_iam(base_conf, target_conf):
+    # Check the update of Iam section with both updatable and not updatable keys
+    for iam_section_child_key, iam_section_child_value in {
+        "ResourcePrefix": ["/path/name-prefix", UpdatePolicy.UNSUPPORTED],
+        "PermissionBoundary": ["arn:aws:iam::aws:policy/some_old_permission_boundary", UpdatePolicy.SUPPORTED],
+        "Role": [{"LambdaFunctionsRole": "arn:aws:iam::aws:role/some_old_role"}, UpdatePolicy.SUPPORTED],
+    }.items():
+        base_conf.update(
+            {
+                "Iam": {
+                    f"{iam_section_child_key}": f"{iam_section_child_value[0]}",
+                }
+            }
+        )
+        _check_patch(
+            base_conf,
+            target_conf,
+            [
+                Change(
+                    ["Iam"],
+                    f"{iam_section_child_key}",
+                    f"{iam_section_child_value[0]}",
+                    None,
+                    UpdatePolicy(
+                        iam_section_child_value[1],
+                    ),
+                    is_list=False,
+                )
+            ],
+            iam_section_child_value[1],
```

Comment:
minor comment to improve readability: let's make more explicit that `iam_section_child_value[1]` contains the expected outcome of the update.
We can do that in two ways:
1. declare a variable within the loop body `expected_patch_outcome=iam_section_child_value[1]`
2. make every value in the dictionary you're iterating on a dictionary rather than an array, e.g. 
```
{
        "ResourcePrefix": dict(name=/path/name-prefix",expected_update_outcome=UpdatePolicy.UNSUPPORTED)
    }
```
3. use named arguments for the function `_check_patch()`

I know that we did not do this improvement elsewhere, but it would make the code reading/reviewing easier.

Reply:
Sure.